### PR TITLE
Send points with API call

### PR DESCRIPTION
**Pull request by Fraseer**
PT Link: https://www.pivotaltracker.com/story/show/180227825

## User Story
As an API
In order to tell the client which departments are taking part in FikaTogether,
I would like to include information on averaged FikaPoints when returning the department

## Changes Proposed
- Update Department Serializer to return averaged fika score & number of participants
- Create method for averaged Fika Score
- FIka score = Total sum of participants scores within a department /  number of participants within a department

## Screenshots
image

image

### CODE REVIEW

File: db/migrate/20211124202133_add_average_score_.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class AddAverageScore < ActiveRecord::Migration[6.1]
+  def change
+    add_column :departments, :average_score, :integer, default: 0
```
Comment (maxarvid) [inline db/migrate/20211124202133_add_average_score_.rb]: We can discuss, but I would suggest making `average_score` a float, then passing along only two decimal points to the client.

File: app/models/department.rb [hunk 2]
```diff
@@ -1,4 +1,22 @@
...
+      department.number_of_participants = department.participants.count
+      department.save
+    end
+  end
+
+  def self.calculate_average_score
+    departments = Department.all
+    departments.each do |department|
+      individual_scores = department.participants.pluck(:fika_score)
+      num_participants = department.participants.count
```
Comment (maxarvid) [inline app/models/department.rb]: Would `department` not have the `number_of_participants` at this point? If that function is always run first in the controller, we might be able to delete this line

Comment (Fraseer) [reply app/models/department.rb]: It would indeed

File: db/migrate/20211124202133_add_average_score_.rb (hunk 1 above)
Comment (Fraseer) [reply db/migrate/20211124202133_add_average_score_.rb]: Good idea, updated to that

**Review by maxarvid — APPROVED** [review]
LvGTM. v for very

File: app/models/fika.rb [hunk 3]
```diff
@@ -24,10 +24,10 @@ def create_calendar_entry
...
     participant_1 = Fika.last.participants[0]
     participant_2 = Fika.last.participants[1]
-    
-    participant_1.fika_score =+ Fika.last.score[0]
-    participant_2.fika_score =+ Fika.last.score[1]
-    
+
+    participant_1.fika_score += Fika.last.score[0]
+    participant_2.fika_score += Fika.last.score[1]
+
```
Comment (dorianbuck) [inline app/models/fika.rb:30]: Nice save!

**Review by dorianbuck — APPROVED** [review]

**Pull request merged**